std/file.d

```diff
@@ -199,120 +245,50 @@ void[] read(in char[] name, size_t upTo = size_t.max) @safe
-                HANDLE.init);
-        auto h = trustedCreateFileW(name, defaults);
+    static trustedCreateFileW(in char[] fileName, DWORD dwDesiredAccess, DWORD dwShareMode,
+                              SECURITY_ATTRIBUTES *lpSecurityAttributes, DWORD dwCreationDisposition,
+                              DWORD dwFlagsAndAttributes, HANDLE hTemplateFile) @trusted
+    {
+        return CreateFileW(fileName.tempCStringW(), dwDesiredAccess, dwShareMode,
+                           lpSecurityAttributes, dwCreationDisposition,
+                           dwFlagsAndAttributes, hTemplateFile);
 
-        cenforce(h != INVALID_HANDLE_VALUE, name);
-        scope(exit) cenforce(trustedCloseHandle(h), name);
-        auto size = trustedGetFileSize(h, null);
-        cenforce(size != INVALID_FILE_SIZE, name);
-        size = min(upTo, size);
-        auto buf = uninitializedArray!(ubyte[])(size);
-        scope(failure) delete buf;
-
-        DWORD numread = void;
-        cenforce(trustedReadFile(h,buf.ptr, size, trustedRef(numread), null) != 0
-                && numread == size, name);
-        return buf[0 .. size];
     }
-    else version(Posix)
+    static trustedCloseHandle(HANDLE hObject) @trusted
     {
-        import core.memory;
-        // A few internal configuration parameters {
-        enum size_t
-            minInitialAlloc = 1024 * 4,
-            maxInitialAlloc = size_t.max / 2,
-            sizeIncrement = 1024 * 16,
-            maxSlackMemoryAllowed = 1024;
-        // }
-
-        static trustedOpen(in char[] path, int oflag) @trusted
-        {
-            return core.sys.posix.fcntl.open(path.tempCString(), oflag);
-        }
-        static trustedFstat(int path, stat_t* buf) @trusted
-        {
-            return fstat(path, buf);
-        }
-        static trustedRead(int fildes, void* buf, size_t nbyte) @trusted
-        {
-            return core.sys.posix.unistd.read(fildes, buf, nbyte);
-        }
-        static trustedRealloc(void* p, size_t sz, uint ba = 0, const TypeInfo ti = null) @trusted
-        {
-            return GC.realloc(p, sz, ba, ti);
-        }
-        static trustedPtrAdd(void[] buf, size_t s) @trusted
-        {
-            return buf.ptr+s;
-        }
-        static trustedPtrSlicing(void* ptr, size_t lb, size_t ub) @trusted
-        {
-            return ptr[lb..ub];
-        }
-
-        immutable fd = trustedOpen(name,
-                core.sys.posix.fcntl.O_RDONLY);
-        cenforce(fd != -1, name);
-        scope(exit) core.sys.posix.unistd.close(fd);
-
-        stat_t statbuf = void;
-        cenforce(trustedFstat(fd, trustedRef(statbuf)) == 0, name);
+        return CloseHandle(hObject);
+    }
+    static trustedGetFileSize(HANDLE hFile, DWORD *lpFileSizeHigh) @trusted
+    {
+        return GetFileSize(hFile, lpFileSizeHigh);
+    }
+    static trustedReadFile(HANDLE hFile, void *lpBuffer, DWORD nNumberOfBytesToRead,
```

Comment:
@yebblies I didn't change that code, just moved it around. It is utter drivel and needs to be removed.